### PR TITLE
fix: refactor system-e2e for cypress v10 upgrade

### LINKED ISSUE #7875: Stabilisation: upgrade Cypress to v10

### PR DESCRIPTION
**Pull request by busla**
closes #7875 
## What

Origin and session beta feature in Cypress v9 might be causing test timeouts

## Why

The repo is using Cypress v9 but v10 vas released a while ago with several bug fixes and improvements to OriginAndSessionFeature

This PR upgrades to v10 and due to breaking changes the cognitoLogin and idsLogin custom commands were refactored to meet those changes. Therefore, it contains both a package upgrade and refactored code.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by pshomov — APPROVED** [review]
Ok, let's see if this will fly ;)

**Comment by busla** [comment]
> Ok, let's see if this will fly ;)

upgraded cypress version in Dockerfile and safely ts-ignored the cypress type error: https://github.com/cypress-io/cypress/issues/9571

**Comment by busla** [comment]
Cypress is running the tests with Electron as default browser, we should probably add specify `--browser chrome` here:

https://github.com/island-is/island.is/blob/170216fa46b70d3cfbf03d8abd18091f32ea6366/scripts/e2e-ci.js#L66

File: apps/system-e2e/entrypoint.sh [hunk 1]
```diff
@@ -2,12 +2,12 @@
...
 
-jq --arg testFiles "$TEST_FILES" '. + {testFiles: ($testFiles | split(",") | map(. + "/*.spec.ts"))}' < cypress.json > cypress.json.tmp
-mv cypress.json.tmp cypress.json
+# jq --arg testFiles "$TEST_FILES" '. + {testFiles: ($testFiles | split(",") | map(. + "/*.spec.ts"))}' < cypress.json > cypress.json.tmp
+# mv cypress.json.tmp cypress.json
 
 echo "Current test environment: ${TEST_ENVIRONMENT}"
 
-echo "Using this configuration:"
-cat cypress.json
```
Comment (kristofer-lp8f) [inline apps/system-e2e/entrypoint.sh:11]: Remove if unused?

**Review by kristofer-lp8f — COMMENTED** [review]
Can you please move all the non-upgrade specific changes to a new PR? If you intended to bundle the tests together with this PR, at least mention it in the title and description.

Please fill out the description of the PR explaining what you're doing and why. If this is still in progress, we can put PRs in a draft state.

**Comment by busla** [comment]
good point, thanks 🙌

**Review by kristofer-lp8f — APPROVED** [review]

**Pull request merged**